docs/blocks/creating-dynamic-blocks.md

```diff
@@ -176,3 +176,107 @@ registerBlockType( 'my-plugin/latest-post', {
 {% end %}
 
 The PHP code is the same as above and is automatically handled through the WP REST API.
+
+## Rendering block markup with PHP
+
+Dynamic blocks can be used in order to control the markup of your block with PHP. This is useful as if you decide to change your blocks markup, you won't have to go back and re-save all the posts which use that block, in order for your blocks markup change to be reflected on all posts.
```

Comment:
There are some issues here with grammar (`blocks markup` should be `block’s markup`).

But beyond that, I don't think we should be promoting dynamic blocks in this way… dynamic blocks exist as a way to render content that *requires* interaction with the server, but not as a way to not have to worry about changing block markup or avoiding dealing with deprecating block markup.

This write-up makes it sound like the purpose of dynamic blocks is to make it easy for block developers to change their markup without effort… but that's not their purpose 😄

That approach can be problematic for users too because their content will change without them expecting it when a block is updated.

Reply:
@tofumatt thanks for the feedback - grammar is not my strong point.

The wording here doesn't insist that dynamic blocks can be used for controlling block markup, only that they can be used in this way.

 > Dynamic blocks can be used in order to control the markup of your block with PHP

The key word here is 'can'.

If dynamic blocks shouldn't be used to for controlling block markup, what can please? Saving the markup in the `the_content` means that the markup cannot be changed or updated without the post being re-saved. Obviously, there must be another method for doing this and I have got mixed up between the two.

Reply:
No worries about the grammar! 😄

I suppose if you wanted to centrally control markup you could use dynamic blocks, but my question would be why you'd want to control the markup itself. I can see using dynamic blocks for things like `Latest Posts` because the client can't render them, but the client can render HTML just fine and keeping content local improves the editing experience.

That said, there's (currently) no method to deal with updating block content en masse if it's not dynamic.

My worry is that the docs, as worded, will push developers toward dynamic blocks for ease of creation at the cost of user experience editing content. Dynamic blocks can't re-render offline, will be slower because of network latency, etc. Maybe I'm being a bit too cautious but I know there's a temptation to use dynamic blocks for everything to avoid much JS 😅

Reply:
If the current implementation means that markup cannot be changed without updating all posts that use a block then this is a *massive* problem. Imagine you make a block for a multi-site install of WordPress (we have 2 with over 500 sites on each) and realise that you need to change the markup for whatever reason - it happens more than you think. For example, if the design of the site changes and an additional element is needed to wrap, go before or after content to meet the needs of the design.

Each site might have hundreds of posts, pages etc. using that block and therefore a way to do this without saving the block on all posts is a must here.

Maybe this is a seperate issue that needs addressing?

Reply:
Indeed, that is how Gutenberg works and you're right: it is a separate issue. You can follow the discussion here #8015 and possibly #7604 as well.

I think it would be cool to have a mechanism to update existing blocks in old posts but I don't think we've settled on anything yet.

Reply:
Heh, I see. 😅